### DISCUSSION TITLE
Allow for oblique-drawn tile sets

### DISCUSSION THREAD

**Discussion by technofromage** [godotengine/godot-proposals#6521] [2D]
### Describe the project you are working on
I'm trying to make an isometric 2d tile-based game with cardinal controls. 
I'm using Godot 4.0, and I've tried both stable and 4.0.1-rc1
I'm working in windows 11

### Describe the problem or limitation you are having in your project

While making our game in Godot 3 I've drawn my tile-set with an oblique isometric shape so that "up" and "right where clearly distinct
I used the "custom" mode on the tile-map and the custom transform on the cell section of the tile-map to get them to tile.
Here's a simplified and trimmed version of my tile-set, if my explanation is lacking
![isomap_Tilted](https://user-images.githubusercontent.com/53595974/225779424-e8164c35-390e-4a5b-b262-002be1560a9f.png)

It would be nice to be able to place a tile-set like this in Godot 4; I cannot find a setting to allow me to tile this properly.

This is related to this closed conversation: https://github.com/godotengine/godot-proposals/discussions/5691
The current implementation of the Node2D skew does not work for my specific use-case because I don't want the textures of the tile-set to be skewed like the grid

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I don't know if it would be easier to implement a "custom" shape, keep using the isometric shape with the Node2D skew and allow the textures to remain not-skewed, or if there's some other good option to handle this issue. 

I do not know enough about Godot's internal architecture to confidently suggest anything as the right answer

### If this enhancement will not be used often, can it be worked around with a few lines of script?:
I understand this use-case is not very common
The work around I see would involve me drawing the tile set via script, either with manual placement math, or trying to override the node2D texture skew. Both of these options seem cumbersome.

### Is there a reason why this should be core and not an add-on in the asset library?
I understand an add-on could be made for this, but this was a feature in Godot 3, so this feels like a feature regression

**Comment by FluffyGroovy** [comment]
I have the same problem.

I'm working on a grid-based game that uses Civ-style angled isometric hex tiles. I created those using a custom transform in Godot 3. In Godot 4, I have to use the skew property. This makes it easier to draw the grid - it takes regular hex images and transforms them - but it also stretches the artwork and makes it look like a 2D object in 3D space.

Godot 3 using the custom transform:
![image](https://github.com/godotengine/godot-proposals/assets/12816703/b86ee759-99f9-4208-b9e3-ed680e07324f)

Godot 4 using the skew property:
![image](https://github.com/godotengine/godot-proposals/assets/12816703/5cae55b3-a8ee-4834-85f7-cbb6e27192bc)

**Comment by scarzehd** [comment]
I'm also trying to implement something similar. I want a tileset where the top and bottom are horizontal but the sides are both slanted to the right. This doesn't seem to be possible anymore without messing up pixel sizing on low-resolution pixel art.

**Comment by Mickeon** [comment]
I think this warrants for something more modular, like the 3.x TileMap's custom mode you mentioned. It would definitely be more encompassing.
